src/main/java/org/dbos/apiary/mysql/MysqlContext.java

```diff
@@ -63,7 +65,7 @@ private void prepareStatement(PreparedStatement ps, Object[] input) throws SQLEx
         }
     }
 
-    public void executeUpsert(String tableName, String id, Object... input) throws Exception {
+    public void executeUpsert(String tableName, String id, String visbilityUpdatePredicate, Object... input) throws Exception {
```

Comment:
What does this new parameter do?  I assume it's debugging code.  If it isn't, it shouldn't be exposed in the user interface.